src/domain/Filters/AddFilterDropdownMenu/AddFilterDropdownMenu.tsx

```diff
@@ -24,129 +24,179 @@ interface IAddFilterDropdownMenuProps {
   componentContext?: string
 }
 
-interface IAddFilterDropdownMenuFilter {
-  type: 'SINGLE_SELECT' | 'NUMBER'
+interface IAddFilterDropdownMenuFilter<FilterValue = string | number> {
+  /** Internal name of the filter, so it can be distinguished from other filters */
   fieldName: string
+  /** Human readable name used to show the filter */
   label: string
-  selectOptions: ISelectOption[]
+  /** Text used to describe how the filter is applied. Defaults to 'is' */
+  operator?: string
+  /** List of select options to fill the select input for the filter with */
+  selectOptions?: Array<ISelectOption<FilterValue>>
+  /** A custom input component to use to select the filter. Overrides selectOptions */
+  customInputComponent?: React.ComponentType<{ onChange: (value: FilterValue | null) => void, value: FilterValue | null}>
+  /**
+   * @deprecated was never fully implemented and custom input component covers it's cases
+   */
+  type?: 'SINGLE_SELECT' | 'NUMBER'
```

Comment:
Can probably be dropped entirely, the ts types are probably compatible as this will be a subset

Reply:
won't it fail if you are doing something like 
```ts
const value: IAddFilterDropdownMenuFilter[] = [
  fieldName: 'blah',
  label: 'also blah',
  type: 'SINGLE_SELECT'
```

Reply:
hmm, you're right. I'd update the deprecation message to just be `@deprecated This is unused and has no effect`